core/src/main/java/org/apache/iceberg/ClientPool.java

```diff
@@ -36,7 +36,7 @@
   private volatile int currentSize;
   private boolean closed;
 
-  ClientPool(int poolSize, Class<? extends E> reconnectExc) {
+  protected ClientPool(int poolSize, Class<? extends E> reconnectExc) {
```

Comment:
nit: unnecessary keyword `protected`

Reply:
I don't think this is unnecessary. This base class is public now and may be extended outside of this package.